File: engine/src/main/java/com/mjaseem/trick/strategy/PromptingStrategy.java
package com.mjaseem.trick.strategy;

import com.mjaseem.trick.engine.Card;
import com.mjaseem.trick.engine.GameHistory;
import com.mjaseem.trick.engine.Records;
import com.mjaseem.trick.engine.Trick;

import java.util.List;

public class PromptingStrategy implements Strategy {
    int nextPlay = Records.MAX_CARDS;

    @Override
    public int chooseCard(List<Card> hand, GameHistory history, Trick trick, Records.Suit trumpSuit) {
        int tmp = this.nextPlay;
        this.nextPlay = Records.MAX_CARDS;
        return tmp;
    }

    public boolean shouldWait() {
        return nextPlay == Records.MAX_CARDS;
    }

    public void setNextPlay(int nextPlay) {
        this.nextPlay = nextPlay;
    }
}

File: rl/strategy/prompting_strategy.py
from engine import constants, Strategy


class PromptingStrategy(Strategy):

    def __init__(self):
        self.next_play = constants.MAX_CARD_RANK

    def choose_card(self, hand, history, trick, trump_suit):
        tmp = self.next_play
        self.next_play =  constants.MAX_CARD_RANK
        return tmp

    def should_wait(self):
        return self.next_play ==  constants.MAX_CARD_RANK

    def set_next_play(self, next_play):
        self.next_play = next_play
